netlogo-gui/src/main/agent/Torus3D.java

```diff
@@ -8,51 +8,59 @@
 public final strictfp class Torus3D
     extends Torus
     implements Topology3D {
+
+  // this class *ought* to be able to see `world` in Topology, but
+  // it can't, probably due to the scala class `Torus` between this
+  // class and Topology. in the meantime, adding this field is good enough
+  // for now.
+  final World _world;
```

Comment:
So, the following breaks:
```java
abstract class A {
  final String s;

  A(String string) {
    s = string;
  }
}

class C extends B {
  C(String string) {
    super(string);
  }

  public String foo() {
    return s;
  }
}
```

```scala
class B(s: String) extends A(s) {
  def bar = s
}
```
But if the method `bar` is removed from `B` so that `B` never references `s`, it's fine. Alternatively, it's also fine if `B`'s constructor parameter is named `_s`. I'm not quite sure how this actually ever worked, as my minimal test case failed compilation in the same way in 2.11.7, 2.11.5, and 2.12.0. I've decided that rather than adding this field, it's easiest to rename the constructor parameter in `Torus`. I'm happy to open a bug for this if you would like.